video/audio_channel.h

```diff
@@ -129,85 +135,104 @@ uint8_t AudioChannel::getStatus() {
 	return status;
 }
 
-std::unique_ptr<fabgl::WaveformGenerator> AudioChannel::getSampleWaveform(uint16_t sampleId, std::shared_ptr<AudioChannel> channelRef) {
+std::shared_ptr<fabgl::WaveformGenerator> AudioChannel::getSampleWaveform(uint16_t sampleId, std::shared_ptr<AudioChannel> channelRef) {
 	if (samples.find(sampleId) != samples.end()) {
 		auto sample = samples.at(sampleId);
+		// if (sample->channels.find(_channel) != sample->channels.end()) {
+		// 	// this channel is already playing this sample, so do nothing
+		// 	debug_log("AudioChannel: already playing sample %d on channel %d\n\r", sampleId, channel());
+		// 	return nullptr;
+		// }
+
+		// // if we're already a waveform, then swap sample
+		// if (this->_waveformType == AUDIO_WAVE_SAMPLE) {
+		// 	// swap sample
+		// 	_mutex.lock();
+		// 	((EnhancedSamplesGenerator *)this->_waveform.get())->setSample(sample);
+		// 	_mutex.unlock();
+		// 	debug_log("AudioChannel: setSample %d on channel %d\n\r", sampleId, channel());
+		// 	return nullptr;
+		// }
```

Comment:
swapping out just the sample when the channel is already using an EnhancedSampleGenerator in principle makes this more efficient.

this however also appears to make things more unstable and more likely to cause an issue - so for now this is disabled

Reply:
this has been removed

swapping the AudioSample object being used by a waveform generator isn't reliable - it's fairly cheap to make a new generator, so we may as well do that